apple/swift-docc
# Check symbols are deprecated on at least one platform before marking unconditionally deprecated

Bug/issue #, if applicable: rdar://112197357

## Summary

Currently, DocC checks whether a symbol is either deprecated or unavailable for all platforms it is aware of to determine whether a symbol is unconditionally deprecated.

However, if a symbol is marked unavailable on all the platforms DocC has availability information about, this check causes DocC to mark the symbol unconditionally deprecated.

This PR adds a check that the symbol is deprecated on at least one platform to prevent symbols that are marked _unavailable_ from being displayed as unconditionally _deprecated_.

## Testing

Steps:
1. Create a project that does not give DocC any default platform availability information.
2. Mark a symbol as unavailable.
3. Build documentation and check that the symbol is not deprecated.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
